### PR TITLE
CTECH-3411:

### PR DESCRIPTION
**Pull request by adamMQM**
# Pull Request Checklist

- [x] Read the [contributing guidelines](../blob/master/docs/CONTRIBUTING.md)
- [x] Tests pass
- [x] Raised the PR against the `develop` branch

# Description of the PR

Add IHttpClientFactory interface for configurable injection of HttpClient.
Move logic from ConfigureMessageHandler to TcpKeepAliveHttpClientFactory. 
Remove duplicate configuration of HttpMessageHandler by calling RestClient constructor with HttpClient.
Add HttpClientFactory to ReadableConfiguration and configuration objects. 
By default, use TcpKeepAliveHttpClientFactory in GlobalConfiguration

### CODE REVIEW

File: generate/templates/IReadableConfiguration.mustache
Comment (hanialsibai) [inline generate/templates/IReadableConfiguration.mustache:1]: What's the difference between the default_templates and templates? This and configuration is in both

Comment (adamMQM) [reply generate/templates/IReadableConfiguration.mustache:1]: Default templates are the original templates supplied by openapi - which we modify. We supply them in default_templates so they can be diffed with the modified version.

File: generate/default_templates/Configuration.mustache [hunk 1]
```diff
@@ -702,6 +702,7 @@ namespace {{packageName}}.Client
                 TempFolderPath = second.TempFolderPath ?? first.TempFolderPath,
                 DateTimeFormat = second.DateTimeFormat ?? first.DateTimeFormat,
                 ClientCertificates = second.ClientCertificates ?? first.ClientCertificates,
+                HttpMessageHandler =  second.HttpMessageHandler ?? first.HttpMessageHandler,
```
Comment (AndrewNeudegg) [inline generate/default_templates/Configuration.mustache:705]: Why do we merge configurations here? What benefit does it have?

File: generate/templates/ApiClient.mustache [hunk 2]
```diff
@@ -525,7 +529,10 @@ namespace {{packageName}}.Client
                     configuration);
             }
 
-            RestClient client = new RestClient(clientOptions,
+           var httpClient = new HttpClient(_createHttpMessageHandler(clientOptions), _disposeHandler);
```
Comment (AndrewNeudegg) [inline generate/templates/ApiClient.mustache:532]: indentation -- do we have a linter?

**Review by AndrewNeudegg — APPROVED** [review]

**Review by AndrewNeudegg — APPROVED** [review]

**Pull request merged**